apidef/graphql_config_adapter.go

```diff
@@ -0,0 +1,280 @@
+package apidef
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	"github.com/jensneuse/graphql-go-tools/pkg/ast"
+	"github.com/jensneuse/graphql-go-tools/pkg/astparser"
+	graphqlDataSource "github.com/jensneuse/graphql-go-tools/pkg/engine/datasource/graphql_datasource"
+	"github.com/jensneuse/graphql-go-tools/pkg/engine/datasource/httpclient"
+	restDataSource "github.com/jensneuse/graphql-go-tools/pkg/engine/datasource/rest_datasource"
+	"github.com/jensneuse/graphql-go-tools/pkg/engine/plan"
+	"github.com/jensneuse/graphql-go-tools/pkg/graphql"
+)
+
+var ErrUnsupportedGraphQLConfigVersion = errors.New("provided version of GraphQL config is not supported for this operation")
+
+type GraphQLConfigAdapter struct {
+	config     GraphQLConfig
+	httpClient *http.Client
+}
+
+func NewGraphQLConfigAdapter(config GraphQLConfig) GraphQLConfigAdapter {
+	return GraphQLConfigAdapter{config: config}
+}
+
+func (g *GraphQLConfigAdapter) EngineConfigV2() (*graphql.EngineV2Configuration, error) {
+	if g.config.Version != GraphQLConfigVersion2 {
```

Comment:
we could do this check in `NewGraphQLConfigAdapter` and we will know immediately is adapter capable to handle config

Reply:
this is similar to the previous comment: As an adapter is for generating different outputs it could be, that one output will need a `GraphQLConfigVersion3`for example - so the check should live outside `NewGraphQLConfigAdapter`